cmd/soroban-cli/src/events.rs

```diff
@@ -0,0 +1,192 @@
+        }
+
+        if self.start_ledger > self.end_ledger {
+            return Err(Error::InvalidLedgerRange {
+                low: self.start_ledger,
+                high: self.end_ledger,
+            });
+        }
+
+        for raw_contract_id in &self.contract_ids {
+            // We parse the contract IDs to ensure they're the correct format,
+            // but since we'll be passing them as-is to the RPC server anyway,
+            // we disregard the return value.
+            utils::contract_id_from_str(raw_contract_id).map_err(|e| {
+                Error::CannotParseContractId {
+                    contract_id: raw_contract_id.clone(),
+                    error: e,
+                }
+            })?;
+        }
+
+        let mut events: Vec<Event> = Vec::new();
+        if self.rpc_url.is_some() {
+            let client = Client::new(self.rpc_url.as_ref().unwrap());
+            let rpc_event = client.get_events(&self.contract_ids, &self.topics).await?;
+            events = rpc_event.events;
+        } else if self.sandbox.is_some() {
+            // TODO: Get events from the sandbox.
+            let path = self.sandbox.as_ref().unwrap();
+            if !path.exists() {
+                return Err(Error::InvalidSandboxFile {
+                    path: path.to_str().unwrap().to_string(),
+                });
+            }
+        } else {
+            return Err(Error::TargetRequired {});
+        }
+
+        for event in &events {
+            if self.format == "json" {
+                println!("{}", serde_json::to_string_pretty(&event).unwrap());
+            } else {
+                print_event_in_color(event)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+pub fn print_event_in_color(event: &Event) -> Result<(), Box<dyn std::error::Error>> {
+    let mut stdout = StandardStream::stdout(ColorChoice::Auto);
+
+    set_white(&mut stdout)?;
+    write!(&mut stdout, "Event ")?;
+    set_green(&mut stdout)?;
+    write!(&mut stdout, "{}", event.id)?;
+
+    set_white(&mut stdout)?;
+    write!(&mut stdout, ":\n  Ledger:   ")?;
+    set_green(&mut stdout)?;
+    write!(&mut stdout, "{}", event.ledger)?;
+    set_white(&mut stdout)?;
+    write!(&mut stdout, " (closed at ")?;
+    set_green(&mut stdout)?;
+    write!(&mut stdout, "{}", event.ledger_closed_at)?;
+
+    set_white(&mut stdout)?;
+    write!(&mut stdout, ")\n  Contract: ")?;
+    set_green(&mut stdout)?;
+    write!(&mut stdout, "{}", event.contract_id)?;
+
+    set_white(&mut stdout)?;
+    write!(&mut stdout, "\n  Topics:")?;
+    set_green(&mut stdout)?;
+    for topic in event.topic.iter() {
+        let scval = ScVal::from_xdr_base64(topic)?;
+        write!(&mut stdout, "\n            {:?}", scval)?;
+    }
+    set_white(&mut stdout)?;
+    write!(&mut stdout, "\n  Value: ")?;
+    set_green(&mut stdout)?;
+    let scval = ScVal::from_xdr_base64(&event.value)?;
+    writeln!(&mut stdout, "{:?}", scval)?;
+
+    set_white(&mut stdout)?;
```

Comment:
Can just return the result.
```suggestion
    set_white(&mut stdout)
```

Reply:
This one gave me trouble, since it's an `io::Error` rather than a `Box` error? Which is confusing because it works with the `?` syntax...

Reply:
I think what you're doing here is fine with `?;`.

The `?` syntax converts the `io::Error` into an `Error`. It automatically calls `.into()` on the error. Whereas when you just return the function there's no auto-conversion.